src/Block/Adminhtml/Order/View/View.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * Copyright Shopgate Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+declare(strict_types=1);
+
+namespace Shopgate\Import\Block\Adminhtml\Order\View;
+
+use Magento\Backend\Block\Template;
+use Magento\Framework\Exception\LocalizedException;
+use Shopgate\Base\Api\Data\OrderInterface;
+use Shopgate\Base\Helper\Encoder;
+use Shopgate\Base\Model\Shopgate\Order;
+use Shopgate\Base\Model\Shopgate\OrderRepository;
+use Shopgate\Import\Block\Adminhtml\Order\DataHydrator;
+
+class View extends Template
+{
+
+    /**
+     * @var OrderRepository
+     */
+    private $orderRepository;
+    /**
+     * @var array
+     */
+    private $whitelist;
+    /**
+     * @var Encoder
+     */
+    private $jsonDecoder;
+
+    /**
+     * @param OrderRepository $orderRepository
+     * @param Template\Context $context
+     * @param Encoder $jsonDecoder
+     * @param array $whitelist - our whitelisted data to print, see di.xml
+     * @param array $data
+     */
+    public function __construct(
+        OrderRepository $orderRepository,
+        Template\Context $context,
+        Encoder $jsonDecoder,
+        array $whitelist = [],
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->orderRepository = $orderRepository;
+        $this->whitelist = $whitelist;
+        $this->jsonDecoder = $jsonDecoder;
+    }
+
+    /**
+     * Checks if it's a Shopgate order
+     *
+     * @return bool
+     */
+    public function isShopgateOrder(): bool
+    {
+        return $this->getShopgateOrder() && !$this->getShopgateOrder()->isEmpty();
```

Comment:
This might be optimized/cached internally but worst case it would trigger two DB queries?

Reply:
wow, I was so convinced it's not a DB query despite coding it up. My brain is in SW6 code apparently...

Reply:
So it's not only me, that's a relief 😅